service/history/transferQueueProcessor.go

```diff
@@ -114,7 +109,7 @@ func newTransferQueueProcessor(shard ShardContext, visibilityMgr persistence.Vis
 	return processor
 }
 
-func newAckManager(processor transferQueueProcessor, shard ShardContext, executionMgr persistence.ExecutionManager,
+func newAckManager(processor *transferQueueProcessorImpl, shard ShardContext, executionMgr persistence.ExecutionManager,
```

Comment:
Because Impl has the config object in it which is needed by the ackManager. Initially I changed the constructor to accept a config object as well but it required changes in many places so I determined that it wasn't worth it.